### PR TITLE
fix(pii): split compiler-free redaction extra

### LINKED ISSUE #134: pipeline-core is uninstallable on Linux without a C compiler: numpy<2 has no cp313 wheel
Found tonight when the redaction backfill could not install its environment on the CPU box.

## What happens

`uv sync --extra pipeline-core` fails on the box with:

```
numpy==1.26.4 ... Unknown compiler(s): [cc, gcc, clang, ...]
```

## Why

- `pyproject.toml:52` pins `numpy<2` on **pipeline-core**, resolving to `numpy==1.26.4`
- The repo pins `.python-version = 3.13`
- `numpy==1.26.4` predates Python 3.13 and publishes **no cp313 wheel**. `uv.lock` lists **zero** wheels for it — sdist only

So on every platform this builds from source. It succeeds on macOS because Xcode ships clang, and fails on the box because it has no compiler. The pin's own comment says it follows the DSI repo (`numpy<2, opencv<4.12, transformers>=4.57,<5`) — it is inherited, not derived from what these stages need.

**This is not box-specific.** Any Linux machine without build tooling cannot install this extra: the GPU box in week 2, a fresh CI runner, a new contributor on Linux.

## The PII path does not need the pin

Evidence from this machine, running the merged `#69` code:

- local venv has **numpy 2.5.0**
- `presidio_analyzer`, `spacy` and `en_core_web_sm` all import
- `tests/test_redact_grievance.py` — **18 passed, 0 skipped**, including `test_real_redactor_removes_a_mobile_number`, which exercises the real analyzer rather than the stub

So redaction runs correctly on numpy 2.x.

## What the pin is probably protecting

Not redaction. The likely reasons are the other pipeline-core stages:

- the pickled estimators (`page_split` pkl, the label encoder), pinned alongside `scikit-learn>=1.8,<1.9` to "match the era of the pickled estimators" — unpickling across a numpy major is the classic breakage
- `opencv-python<4.12`
- `xgboost` / `scikit-image` expectations

Whoever fixes this should establish which of those actually constrain numpy, rather than assuming the whole extra needs it.

## Options

1. **Split the extra.** A `pii` extra with just presidio, spacy and the model — no numpy ceiling, all wheels available. Cleanest, and matches the existing design where stages import lazily and an environment installs only what it runs.
2. **Relax `numpy<2`** across pipeline-core, after verifying the pickled estimators still load and the ViT/page-split stages still behave. Bigger blast radius, needs real testing.
3. **Provision build tooling** on every Linux host. Treats the symptom, is slow, and `numpy==1.26.4` may not even compile against the 3.13 C API.

**Option 1 is the demo-week answer**; option 2 is the right end state.

## Related

- #69 — the redaction job that hit this
- #71 — the backfill it blocks

### PR DESCRIPTION
**Pull request by ymohanty**
## Summary

Splits Presidio, spaCy, and the English model into a standalone `pii` extra so the redaction/evaluation path no longer inherits `pipeline-core`'s legacy `numpy<2` constraint.

- keeps `numpy<2` on `pipeline-core` for the non-PII stages;
- pins the standalone `pii` environment to `numpy>=2` and resolves it independently;
- retains the all-stage demo's existing embedded PII stack, while documenting that the OCR-plus-PII bootstrap path remains on its legacy dependency contract;
- updates PII/redaction commands and adds frozen-lock wheel-contract coverage.

## Root cause

`numpy==1.26.4` has no CPython 3.13 wheel, so the old `pipeline-core` redaction path attempted a source build on compiler-free Linux hosts.

## Validation

- `uv lock --check`
- `uv run --extra pii pytest tests/test_pii_extra_contract.py tests/test_pii_redaction.py tests/test_redact_grievance.py` — 106 passed
- `uv run --extra serving --extra pipeline-core pytest` — 868 passed, 14 skipped
- `uv run ruff check .`
- `git diff --check`

The lock contract verifies the frozen `pii` export selects NumPy 2.5.1 and that every selected native dependency has a Linux x86_64 CPython 3.13-compatible wheel.

Fixes #134.

### CODE REVIEW

**Comment by ymohanty** [comment]
@codex review

File: tests/README.md [hunk 1]
```diff
@@ -5,7 +5,9 @@
 SQLite artifact DBs) — not mocks of our own code. Gate before "done":
 
 ```bash
-uv run --extra pipeline-core pytest && uv run ruff check .
+uv run --extra serving --extra pipeline-core pytest
+uv run --extra pii pytest tests/test_pii_extra_contract.py tests/test_pii_redaction.py tests/test_redact_grievance.py
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply tests/README.md]: Fixed in d81541b. The documented PII gate now includes `test_rederive_pii_draft.py` and `test_bootstrap_pii_gold.py` alongside the contract/redaction modules. The complete PII gate ran 124 tests successfully.

File: pyproject.toml [hunk 2]
```diff
@@ -40,11 +40,20 @@ dependencies = [
...
+# on wheels for CPython 3.13; unlike pipeline-core, it has no numpy ceiling.
+pii = [
+    "numpy>=2",
+    "presidio-analyzer>=2.2,<3",
+    "presidio-anonymizer>=2.2,<3",
+    "spacy>=3.8,<3.9",
+    "en-core-web-sm @ https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.8.0/en_core_web_sm-3.8.0-py3-none-any.whl",
+]
+# format_classifier, pytesseract OCR, page_type, summarizer
 pipeline-core = [
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply pyproject.toml:57]: Fixed in d81541b. `pipeline-sample` keeps its original format/OCR/PII scope but now runs sequential `pipeline-core` then `pii` invocations against the same input, models directory, and SQLite artifact DB. A contract test asserts the exact extras, stages, shared paths, and DVC dependencies.

File: README.md [hunk 3]
```diff
@@ -93,15 +93,24 @@ must be quoted with: [janasunani/analytics/README.md](janasunani/analytics/READM
...
   --db data/processed/pipeline.sqlite \
   --models models \
-  --ocr-engine pytesseract          # deepseek needs CUDA (the GPU box)
+  --ocr-engine pytesseract \
+  --stages format_classifier ocr_extraction  # deepseek needs CUDA (the GPU box)
+
+uv run --extra pii janasunani-pipeline run \
+  --db data/processed/pipeline.sqlite \
+  --models models \
+  --stages pii_tagger
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply README.md:114]: Fixed in d81541b. The component workflow now continues after PII: page type and summarization run under `pipeline-core`, then categorization runs under the compatible `categorizer` extra, all against the same artifact DB.

File: docs/ARCHITECTURE.md [hunk 4]
```diff
@@ -113,10 +113,13 @@ migration change): **1,371,288 complaints / 6,556,171 action-history rows**.
...
 
 Python ≥ 3.13, managed by **uv**. Base deps are light; heavy ML stacks live in
-three **optional extras** (`pyproject.toml`):
+four **optional extras** (`pyproject.toml`):
 
-- `pipeline-core` — format classifier, pytesseract OCR, Presidio PII, page-type
-  ViT, summarizer (`transformers>=4.57,<5`)
+- `pipeline-core` — format classifier, pytesseract OCR, page-type ViT,
+  summarizer (`transformers>=4.57,<5`); its inherited `numpy<2` pin remains
+  isolated from the standalone `pii` extra
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply docs/ARCHITECTURE.md:120]: Fixed in d81541b. ROADMAP now defines all four extras, the staged CPU workflow across compatible environments, and the complete standard plus PII validation commands; README, ARCHITECTURE, pipeline docs, and tests docs match it.

**Comment by ymohanty** [comment]
@codex review

File: pyproject.toml [hunk 5]
```diff
@@ -40,11 +40,20 @@ dependencies = [
...
+# installs the extra(s) for the stages it runs. The inherited `numpy<2` pin is
+# kept with the non-PII pipeline stages; Presidio/spaCy run from the small `pii`
+# extra, which can use a Python 3.13 wheel on a compiler-free Linux host.
 [project.optional-dependencies]
-# format_classifier, pytesseract OCR, pii_tagger, page_type, summarizer
+# PII redaction/evaluation only. A numpy 2.x floor keeps this standalone extra
+# on wheels for CPython 3.13; unlike pipeline-core, it has no numpy ceiling.
+pii = [
+    "numpy>=2",
+    "presidio-analyzer>=2.2,<3",
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply pyproject.toml]: Fixed in 094370f. Both standalone `pii` evaluation and `demo` production now pin `presidio-analyzer==2.2.363`. The PII NumPy range is `>=2,<2.5`, matching Presidio's declared ceiling while retaining a Python 3.13-compatible NumPy 2.4.6 wheel. A dependency-contract regression exports both frozen Linux environments and asserts identical Analyzer 2.2.363 resolution. The real-analyzer PII gate passed 125 tests, and all GitHub checks pass.

File: pyproject.toml [hunk 6]
```diff
@@ -40,11 +40,20 @@ dependencies = [
...
-# (numpy<2, opencv<4.12, transformers>=4.57,<5) and the DeepSeek-OCR model
-# card (transformers 4.46 / tokenizers 0.20).
+# installs the extra(s) for the stages it runs. The inherited `numpy<2` pin is
+# kept with the non-PII pipeline stages; Presidio/spaCy run from the small `pii`
+# extra, which can use a Python 3.13 wheel on a compiler-free Linux host.
 [project.optional-dependencies]
-# format_classifier, pytesseract OCR, pii_tagger, page_type, summarizer
+# PII redaction/evaluation only. A numpy 2.x floor keeps this standalone extra
+# on wheels for CPython 3.13; unlike pipeline-core, it has no numpy ceiling.
+pii = [
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply pyproject.toml:49]: Fixed in 094370f. Root `AGENTS.md` now lists the standalone `pii` extra, explains that redaction/evaluation runs separately from `pipeline-core`, and records the complete five-module PII validation gate. This now matches ROADMAP, README, architecture, and tests documentation. Ruff, lock consistency, the PII gate, and all GitHub checks pass.

**Comment by ymohanty** [comment]
@codex review

File: pyproject.toml [hunk 7]
```diff
@@ -107,6 +111,15 @@ categorizer = [
 demo = [
     "janasunani[serving]",
     "janasunani[pipeline-core]",
+    # The all-stage live demo retains its existing in-process PII stack. It
+    # lists it directly because the standalone `pii` extra is intentionally
+    # resolved separately from `pipeline-core` below.
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply pyproject.toml:116]: Fixed in 8f2e365. `janasunani/serving/README.md` now launches the live server with the complete `demo` extra. A deploy contract scans all known documented `janasunani-api-live` commands and rejects any that do not use `uv run --extra demo`; a frozen Linux export also proves the demo environment contains Presidio Analyzer/Anonymizer, spaCy, and `en-core-web-sm`. The deploy suite passed 50 tests, and all GitHub checks pass.

**Comment by ymohanty** [comment]
@codex review

File: dvc.yaml [hunk 8]
```diff
@@ -17,18 +17,23 @@ stages:
...
     cmd: >-
       uv run --extra pipeline-core janasunani-pipeline run
       --input data/raw/documents-sample
       --db data/processed/pipeline-sample.sqlite
       --models models
       --ocr-engine pytesseract
-      --stages format_classifier ocr_extraction pii_tagger
+      --stages format_classifier ocr_extraction
+      --workers 1 &&
+      uv run --extra pii janasunani-pipeline run
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply dvc.yaml:32]: Fixed in ea02638 (rebased onto current main). `dvc.lock` now records the exact split `pipeline-core` then `pii` command from `dvc.yaml`; a YAML contract test asserts command parity. The update was surgical and command-only: no DVC command ran, nothing under `data/` was accessed, and all dependency/output hashes and cached-artifact metadata remain unchanged. Contract tests, Ruff, offline lock consistency, and all GitHub checks pass.

**Comment by ymohanty** [comment]
@codex review
